### PR TITLE
API: transform behaves differently with 'ffill' on DataFrameGroupBy and SeriesGroupBy

### LINKED ISSUE #24211: transform behaves differently with 'ffill' on DataFrameGroupBy and SeriesGroupBy
#### Code Sample

```python
import pandas as pd
data = [['a', 0.0], ['a', float('nan')], ['b', 1.0], ['b', float('nan')]]
df = pd.DataFrame(data, columns=['key', 'values'])
print(df.groupby('key').transform('ffill'))
print(df.groupby('key')['values'].transform('ffill'))
```
#### Problem description

The first print statement produces 

       values
    0     0.0
    1     0.0
    2     1.0
    3     1.0

The second print statement produces

    0    0.0
    1    0.0
    2    0.0
    3    0.0

#### Expected Output

I expected both operations to compute the same values. I regard the first output as the correct one.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.5.2.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.0-139-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.23.4
pytest: 4.0.1
pip: 18.1
setuptools: 40.6.2
Cython: None
numpy: 1.15.4
scipy: 1.1.0
pyarrow: None
xarray: None
IPython: 7.2.0
sphinx: None
patsy: 0.5.1
dateutil: 2.7.5
pytz: 2018.7
blosc: None
bottleneck: None
tables: 3.4.4
numexpr: 2.6.8
feather: None
matplotlib: 3.0.2
openpyxl: None
xlrd: 1.1.0
xlwt: None
xlsxwriter: 0.7.3
lxml: None
bs4: 4.6.3
html5lib: 1.0.1
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: 0.7.0
</details>

### PR DESCRIPTION
**Pull request by ftrihardjo**
- [ ] closes #24211
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
Thanks @ftrihardjo - could you make the title the same as the issue please, and put the issue number in the pull request body (near `closes #`) ?

File: pandas/tests/groupby/transform/test_transform.py [hunk 1]
```diff
@@ -938,6 +938,16 @@ def test_any_all_np_func(func):
     tm.assert_series_equal(res, exp)
 
 
+def test_transform_ffill():
+    # GH 24211
+    data = [["a", 0.0], ["a", float("nan")], ["b", 1.0], ["b", float("nan")]]
+    df = DataFrame(data, columns=["key", "values"])
+    tm.assert_series_equal(
+        df.groupby("key").transform("ffill")["values"],
```
Comment (jreback) [inline pandas/tests/groupby/transform/test_transform.py:946]: can you use
result=
expected=

File: pandas/tests/groupby/transform/test_transform.py [hunk 2]
```diff
@@ -938,6 +938,16 @@ def test_any_all_np_func(func):
...
+        df.groupby("key")["values"].transform("ffill"),
+    )
+
```
Comment (jreback) [inline pandas/tests/groupby/transform/test_transform.py:949]: test both for seriesgropuby & dataframegroupby

File: pandas/tests/groupby/transform/test_transform.py [hunk 3]
```diff
@@ -938,6 +938,16 @@ def test_any_all_np_func(func):
     tm.assert_series_equal(res, exp)
 
 
+def test_transform_ffill():
+    # GH 24211
+    data = [["a", 0.0], ["a", float("nan")], ["b", 1.0], ["b", float("nan")]]
```
Comment (jreback) [inline pandas/tests/groupby/transform/test_transform.py:943]: can you co-locate with other ffill/filling tests (in this file)

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by MarcoGorelli — APPROVED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @ftrihardjo

**Pull request merged**